### PR TITLE
removed the returnvalue of the inherited parse method

### PR DESCRIPTION
**Pull request by gvanbeck**

### CODE REVIEW

**Comment by Ocramius** [comment]
Can you please adapt the version of php-parser that this is targeting in `composer.json`? If there was an upstream BC break, we need to report it too.

**Comment by gvanbeck** [comment]
Yes, here is the composer json:
{
    "name": "nikic/php-parser",
    "description": "A PHP parser written in PHP",
    "keywords": ["php", "parser"],
    "type": "library",
    "license": "BSD-3-Clause",
    "authors": [
        {
            "name": "Nikita Popov"
        }
    ],
    "require": {
        "php": ">=7.0",
        "ext-tokenizer": "*"
    },
    "require-dev": {
        "phpunit/phpunit": "^6.5 || ^7.0"
    },
    "autoload": {
        "psr-4": {
            "PhpParser\\": "lib/PhpParser"
        }
    },
    "bin": ["bin/php-parse"],
    "extra": {
        "branch-alias": {
            "dev-master": "4.1-dev"
        }
    }
}

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "roave/better-reflection",
+    "name": "gvanbeck/better-reflection",
```
Comment (asgrim) [inline composer.json]: Why are you renaming this...?

File: src/Reflection/ReflectionClass.php [hunk 2]
```diff
@@ -168,8 +168,7 @@ public static function createFromNode(
     public function getShortName() : string
     {
         if (! $this->isAnonymous()) {
-            /** @var Node\Identifier $this->node->name */
-            return $this->node->name->name;
+            return (string)$this->node->name;
```
Comment (asgrim) [inline src/Reflection/ReflectionClass.php]: Can you make this a separate PR for this change please? Also what does this change benefit us by?

File: src/Reflection/ReflectionFunctionAbstract.php [hunk 3]
```diff
@@ -359,15 +363,15 @@ public function isGenerator() : bool
      */
     public function getStartLine() : int
     {
-        return $this->node->getStartLine();
+        return $this->node->getAttributes()['startLine'] ?? -1;
```
Comment (asgrim) [inline src/Reflection/ReflectionFunctionAbstract.php]: Why is this changing to not use the API given?

File: src/Reflection/ReflectionProperty.php [hunk 4]
```diff
@@ -169,7 +169,11 @@ public function getModifiers() : int
      */
     public function getName() : string
     {
-        return $this->node->props[$this->positionInNode]->name->name;
+        if(is_object($this->node->props[$this->positionInNode]->name)) {
```
Comment (asgrim) [inline src/Reflection/ReflectionProperty.php]: Again, not sure what the benefit of this is? Could you explain please?

File: src/SourceLocator/Ast/Parser/MemoizingParser.php [hunk 5]
```diff
@@ -30,7 +30,7 @@ public function __construct(Parser $wrappedParser)
     /**
      * {@inheritDoc}
      */
-    public function parse(string $code, ?ErrorHandler $errorHandler = null) : ?array
+    public function parse($code, ?ErrorHandler $errorHandler = null)
```
Comment (asgrim) [inline src/SourceLocator/Ast/Parser/MemoizingParser.php]: What is the signature of the parent here? Why is it complaining about incompatibility exactly? Was there an upstream change that broke this?

File: src/Util/GetFirstDocComment.php [hunk 6]
```diff
@@ -14,7 +14,8 @@
 {
     public static function forNode(NodeAbstract $node) : string
     {
-        foreach ($node->getComments() as $comment) {
+        $comments = $node->getAttributes()['comments'] ?? [];
```
Comment (asgrim) [inline src/Util/GetFirstDocComment.php]: Why changing away from the given API?

File: composer.json (hunk 1 above)
Comment (gvanbeck) [reply composer.json]: Please I was not meaning to do pullrequests. Please decline them all for today

Sorry to bother

**Comment by Ocramius** [comment]
Moved to https://github.com/Roave/BetterReflection/issues/450

Closing as `invalid` here, since the patch has unrelated/invalid changes.
